Soomla/Assets/Plugins/Soomla/Profile/Domain/UserProfile.cs

```diff
@@ -80,48 +80,48 @@ protected UserProfile(Provider provider, string profileId, string username)
 			this.Username = JSONObject.DecodeJsString(jsonUP[PJSONConsts.UP_USERNAME].str);
 			this.ProfileId = JSONObject.DecodeJsString(jsonUP[PJSONConsts.UP_PROFILEID].str);
 
-			if (jsonUP[PJSONConsts.UP_FIRSTNAME]) {
```

Comment:
@vedi, looks like I understand. Do you mean to make this:
`jsonUP[PJSONConsts.UP_FIRSTNAME] != null && jsonUP[PJSONConsts.UP_FIRSTNAME].type != JSONObject.Type.NULL`
?

>What if jsonUP[PJSONConsts.UP_FIRSTNAME] is null? Is it possible? 
It may happen for non-native things or for future providers.